swift/schema.py

```diff
@@ -516,9 +516,22 @@ class OpaqueValueExpr(Expr):
     pass
 
 class OpenExistentialExpr(Expr):
-    sub_expr: Expr | child
-    existential: Expr | child
-    opaque_expr: OpaqueValueExpr | child
+    """ An implicit expression created by the compiler when a method is called on a protocol. For example in
+    ```
+    protocol P {
+      func foo() -> Int
+    }
+    func bar(x: P) -> Int {
+      return x.foo()
+    }
+    `x.foo()` is actually wrapped in an `OpenExistentialExpr` that "opens" `x` replacing it in its subexpression with
+    an `OpaqueValueExpr`.
```

Comment:
Not necessary to do in this PR, but this description makes it sound like we should probably hide `OpenExistentialExpr` from the AST since this will be very unintuitive to people not familiar with the AST 😅.

Reply:
good point!